pkg/services/ngalert/state/manager_private_test.go

```diff
@@ -847,6 +848,9 @@ func TestProcessEvalResults_StateTransitions(t *testing.T) {
 						executeTest(t, r, tc.results, expectedTransitions, true)
 					})
 					t.Run("applyNoDataErrorToAllStates=false", func(t *testing.T) {
+						if stateExec == ngmodels.KeepLast {
+							t.Skip("KeepLastErrState is not supported with applyNoDataErrorToAllStates=false")
+						}
```

Comment:
If I do not miss anything, according to the tests in `manager_test`, it supports this case, and has the same behavior as other mappings. Please correct me if I miss something.

Reply:
Saying it's unsupported isn't really right here yeah. This needs some more thought given to it since the legacy behavior is very much not desired for keep last state.